### PR TITLE
feat: folder selection support in useMediaAsset and file widget

### PR DESCRIPTION
**Pull request by denyskon**
This PR adds a config parameter to allow selecting folders instead of files in media popup.

The required option is added to useMediaAsset hook and the file widget.

### CODE REVIEW

**Comment by denyskon** [comment]
There seem to be some broken tests, but that seems to be the select widget. Maybe some issues from your refactor a few days ago?

**Comment by KaneFreeman** [comment]
Tests are passing in next now.

File: packages/core/src/components/media-library/common/MediaLibrary.tsx [hunk 1]
```diff
@@ -182,7 +183,14 @@ const MediaLibrary: FC<TranslatedProps<MediaLibraryProps>> = ({
    */
   const handleAssetSelect = useCallback(
     (asset: MediaFile) => {
-      if (!canInsert || selectedFile?.key === asset.key || asset.isDirectory) {
+      console.log(forFolder);
```
Comment (KaneFreeman) [inline packages/core/src/components/media-library/common/MediaLibrary.tsx:186]: Should probably remove this

Comment (KaneFreeman) [reply packages/core/src/components/media-library/common/MediaLibrary.tsx:186]: Looks like there are a few throughout the PR.

**Review by KaneFreeman — APPROVED** [review]

File: packages/core/src/components/media-library/common/MediaLibrary.tsx (hunk 1 above)
Comment (denyskon) [reply packages/core/src/components/media-library/common/MediaLibrary.tsx]: yeah, is removed

**Pull request merged**